content/kotlin/concepts/functions/functions_in_kotlin.md

```diff
@@ -0,0 +1,194 @@
+---
+Title: 'Functions in Kotlin'
+Description: 'Functions are the most essential part of kotlin'
+Subjects: 
+  - 'Mobile Development'
+Tags: 
+  - 'Android'
+  - 'Functions'
+  - 'Kotlin'
+CatalogContent: 
+  - 'learn-kotlin'
+---
+
+# Functions in Kotlin
+
+**Functions are an essential part of any Kotlin program.**Functions are a reusable part of code that can be called in any part of the program. The Kotlin program has to start with the `main()` function, or there will be an error. `fun` is used to define a function in Kotlin, but there are also predefined functions like `println `and `print`. Functions are called in the `main()` function.
+
+## Function `main()`
+
+Function `main()` is the entry point to Kotlin programing and also required in every Kotlin program. The `main()` function is meant to include your code that you write and is not called anywhere. Check that your code is between the `main()` function curly brackets `{}`.
+
+```
+fun main() {
+  println("Hello World")
+}
+```
+## Creating functions in Kotlin ###
+
+User defined functions start with the `fun` keyword, followed by the function name, parameters (if there is any) and the function body which is enclosed in curly braces.
+
+```
+fun functionName(arguments): return type {
+  body
+  return statment
+}
+```
++ *fun* is used to define a function.
++ The *functionName* is used to call the function in the program.
++ *Arguments* have to be in the parentheses, arguments are pieces of information fed to the function.
++ *Return type* declares the type of a returning value. It is optional to include.
++ In the *body* contains the instructions.
++ *Return statment* returns the value of the function.
+
+### Arguments
+
+`Argument` is data inserted to function. `Arguments` must have a name and data type of inserted data. Between `argument` and `data type` must be a colon `:`. The arguments are separated by comma `,` .
+```
+fun functionName(argumentName: dataType, argumentName2 : datatype){
```

Comment:
Move the second example to the pseudocode example below that uses it. Also, since the second example is illustrating an actual function, include an actual body to go with it.